.github/workflows/hack.yml

```diff
@@ -0,0 +1,35 @@
+name: Hack
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+jobs:
+  hack:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo
+            target
+          key: ${{ matrix.os }}-hack
```

Comment:
1. Windows and macOS should be added too
2. `${{ matrix.os }}` is an empty variable since no matrix is setup

It may be worth putting this in `ci.yml` which already has all the Windows workarounds / dependency stuff.

Reply:
ah that's what I get for blindly copying another workflow as a template :p

This workflow takes awhile to complete (~7 mins), so I don't really want to add it as another step to `ci`, I think just testing linux should be enough, we already test full builds with Windows/ MacOS.

If we do add Windows/ MacOS support, it may make sense to separate the `ci` setup into a separate action so that both the `ci` and `hack` workflows can use it.